docs/modules/ROOT/pages/access-control.adoc

```diff
@@ -202,3 +134,143 @@ TIP: A recommended configuration is to grant both roles to a secure governance c
+
+image::access-control-multiple.svg[Access Control multiple]
+
+Protocols managing permissions in production systems often require more integrated alternatives to fragmented permissions through multiple `AccessControl` instances. 
+
+Better role management can be achieved with an xref:api:access.adoc#AccessManager[`AccessManager`] instance. Instead of managing each contract's permission separately, AccessManager stores all the permissions in a single contract, making your protocol easier to audit and maintain.
+
+image::access-manager.svg[Access Manager]
+
+The AccessManager is designed around the concept of role:
+
+* Roles are granted to accounts (addresses) following a many-to-many approach. This means that each user can have one or multiple roles and multiple users can have the same role.
+* Access to a restricted function is restricted to one role. A function is defined by one https://docs.soliditylang.org/en/v0.8.20/abi-spec.html#function-selector[function selector] on one contract (called target). 
+
+For a call to be authorized, the caller must bear the role that is assigned to the current function (contract address + function selector).
+
+image::access-manager-functions.svg[Access Manager functions]
+
+=== Using `AccessManager`
+
+OpenZeppelin Contracts provides xref:api:access.adoc#AccessManager[`AccessManager`]. A contract that can be deployed and used out of the box. It sets an initial admin in the constructor who will be allowed to perform management operations.
+
+In order to restrict access to some functions of your contract, you should inherit from the xref:api:access.adoc#AccessManaged[`AccessManaged`] contract provided along with the manager so that the `restricted` modifier is available to protect any externally facing function. Note you will have to set the address of the Access Manager instance (xref:api:access.adoc#AccessManaged-authority--[authority]) in the constructor so the `restricted` modifier can check permissions by calling the correct manager.
+
+Here's a simple example of an xref:tokens.adoc#ERC20[`ERC20`] token that defines a `mint` functionality that is restricted by an xref:api:access.adoc#AccessManager[`AccessManager`]:
+
+```solidity
+include::api:example$access-control/AccessManagedERC20MintBase.sol[]
+```
+
+NOTE: Make sure you fully understand how xref:api:access.adoc#AccessManager[`AccessManager`] works before using it or copy-pasting the examples from this guide.
+
+Once the managed contract has been deployed, it is now under the manager's control, which will get the minter role assigned to an address and also allow the role to call the `mint` function.
```

Comment:
We should point out that we are starting to talk about client-side Javascript code. e.g.:
```suggestion
Once the managed contract has been deployed, it is now under the manager's control. The initial admin can then assign the minter role to an address and also allow the role to call the `mint` function. For example, this is demonstrated in the following Javascript code:
```

Reply:
I'd like to keep it as pseudocode as possible because ideally, we should provide examples in each library (viem/ethers).

I'd rather merge this and then focus on being library specific in all the documentation pages. We also have [this issue](https://github.com/OpenZeppelin/openzeppelin-contracts/issues/4619)

Reply:
After chatting with Zack we decided to ship the Ethers.js example but we're tracking a larger issue to provide actual code snippets in all our documentation.